### ISSUE TITLE
wasm_i8x16_const fails to compile with integer constants

### ISSUE THREAD

**Issue by zeux** [emscripten-core/emscripten#9768]
I would expect this code to compile:
```
	v128_t mbits = wasm_i8x16_const(1,2,4,8,16,32,64,128, 1,2,4,8,16,32,64,128);
```

And yet it doesn't because the type of each constant is recognized as int.

**Comment by zeux** [comment]
The same problem happens with wasm_i32x4_const when one of the constants is an unsigned int (e.g. 0x80402010 in the example above)

**Comment by tlively** [comment]
Thanks, I'll take a look at how we can loosen those restrictions.

**Comment by tlively** [comment]
Fixed by #9789.

**Comment by zeux** [comment]
Thank you! FWIW I think after this change you're going to get an error (which is expected and useful) if the value is out of range as long as you're using C++11 due to narrowing restrictions, e.g. https://gcc.godbolt.org/z/GN-39N
